### PR TITLE
BDD Testing for pisp-demo-server

### PR DESCRIPTION
**Pull request by kkzeng**
- [x] Write out BDD scenarios
- [x] Write out BDD tests

### CODE REVIEW

File: test/features/firestore-consent-handlers.scenario.feature [hunk 1]
```diff
@@ -0,0 +1,46 @@
+Feature: pisp-demo-server server
+
+Scenario: Create Consent
+  Given pisp-demo-server server
```
Comment (eoln) [inline test/features/firestore-consent-handlers.scenario.feature:4]:
```suggestion
  Given pisp-demo-server
```

File: test/features/firestore-consent-handlers.scenario.feature [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  When I create a new Consent
+  Then the status should be updated and a request id should be assigned
+
+Scenario: Create Consent With Defined Status
+  Given pisp-demo-server server
+  When I create a Consent with a defined status
```
Comment (eoln) [inline test/features/firestore-consent-handlers.scenario.feature:10]:
```suggestion
  When I create a Consent with a "defined" status
```
take a look for reusable snippets and passing parameters: https://github.com/bencompton/jest-cucumber/blob/HEAD/docs/ReusingStepDefinitions.md

**Review by eoln — CHANGES_REQUESTED** [review]
please read documentation: https://github.com/bencompton/jest-cucumber
you can reuse step definitions as described here:  https://github.com/bencompton/jest-cucumber/blob/HEAD/docs/ReusingStepDefinitions.md

File: test/features/firestore-consent-handlers.feature [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  Given pisp-demo-server
+  When I create a Consent with an existing status
+  Then the server should do nothing
+
+Scenario: Create New Consent
+  Given pisp-demo-server
+  When a new Consent is created
+  Then the server should assign a consentRequestId and a new status in the consent repository
+
+Scenario Outline: Update Consent With <Status> Status
```
Comment (eoln) [inline test/features/firestore-consent-handlers.feature:13]: @lewisdaly @jgeewax @kleyow @sridharvoruganti @chbhernani   @elnyry-sam-k 
`Examples:` are a great improvement in BDD testing :-D

File: test/mockData.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Mojaloop Foundation
+ - Name Surname <name.surname@mojaloop.io>
+
+ * Google
+ - Kenneth Zeng <kkzeng@google.com>
+ --------------
+ ******/
+// TODO: Interface wrongly defines request
```
Comment (eoln) [inline test/mockData.ts:25]: do we have a ticket to fix this interface bug, if not please create a one?

**Review by eoln — APPROVED** [review]
lgtm

File: test/mockData.ts (hunk 4 above)
Comment (kkzeng) [reply test/mockData.ts:25]: I created one:
https://app.zenhub.com/workspaces/mojaloop-project-59edee71d1407922110cf083/issues/mojaloop/project/1768

Comment (kkzeng) [reply test/mockData.ts:25]: And another one for amount regexp: https://app.zenhub.com/workspaces/mojaloop-project-59edee71d1407922110cf083/issues/mojaloop/project/1769

**Pull request merged**